src/cascadia/TerminalControl/TSFInputControl.cpp

```diff
@@ -366,14 +362,11 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         {
             const auto incomingText = args.Text();
             const auto range = args.Range();
-
-            _inputBuffer = _inputBuffer.replace(
-                range.StartCaretPosition,
-                range.EndCaretPosition - range.StartCaretPosition,
-                incomingText);
+            const auto [beg, len] = _convertRangeIntoBeginAndLength(range);
+            _inputBuffer.replace(beg, len, incomingText);
             _selection = args.NewSelection();
             // GH#5054: Pressing backspace might move the caret before the _activeTextStart.
-            _activeTextStart = std::min(_activeTextStart, _inputBuffer.size());
+            _activeTextStart = std::min(_activeTextStart, beg);

```

Comment:
This reverts part of ed800dc72de9e51201becf1f752093f4c1b4e155. It still seems to work.